testsuite/src/test/java/org/jboss/intersmash/tests/wildfly/web/cache/offload/infinispan/Infinispan2ReplicasCustomCertificateService.java

```diff
@@ -0,0 +1,137 @@
+/**
+* Copyright (C) 2025 Red Hat, Inc.
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+*         http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+package org.jboss.intersmash.tests.wildfly.web.cache.offload.infinispan;
+
+import io.fabric8.kubernetes.api.model.Secret;
+import io.fabric8.kubernetes.api.model.SecretBuilder;
+import org.apache.commons.io.FileUtils;
+import org.apache.commons.io.IOUtils;
+import org.infinispan.v1.Infinispan;
+import org.infinispan.v1.InfinispanBuilder;
+import org.infinispan.v1.infinispanspec.security.EndpointEncryption;
+import org.infinispan.v1.infinispanspec.security.EndpointEncryptionBuilder;
+import org.infinispan.v2alpha1.Cache;
+import org.jboss.intersmash.application.openshift.OpenShiftApplication;
+import org.jboss.intersmash.application.operator.InfinispanOperatorApplication;
+import org.jboss.intersmash.tests.wildfly.web.cache.offload.infinispan.util.InfinispanCommandLineBasedKeystoreGenerator;
+
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.ArrayList;
+import java.util.Base64;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * Application descriptor that represents an Infinispan/Red Hat Data Grid service deployed by the related Operator, and
+ * having two replicas.
```

Comment:
Here you may want to add that this version is using a custom certificate to encrypt the communication with the endpoint.